From 5f889eaacd901ee74a44da07a28abfb6fbbbbf87 Mon Sep 17 00:00:00 2001
From: George Sapkin <george@sapk.in>
Date: Fri, 23 Jan 2026 20:07:53 +0200
Subject: [PATCH] imagebuilder: fix parsing ABI for apk packages

Fix parsing ABI when package has multiple tags and apk returns them in a
single line.

Fixes: 31cdd13d ("imagebuilder: add ABI suffix to packages when using apk")
Signed-off-by: George Sapkin <george@sapk.in>
Link: https://github.com/openwrt/openwrt/pull/21660
Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 target/imagebuilder/files/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/target/imagebuilder/files/Makefile b/target/imagebuilder/files/Makefile
index c9de9aee93aadc..2b3c0e6ae0671f 100644
--- a/target/imagebuilder/files/Makefile
+++ b/target/imagebuilder/files/Makefile
@@ -157,7 +157,7 @@ endif
 #
 # 1: package name
 define GetABISuffix
-$(shell $(APK) query --fields tags --match provides $(1) | grep openwrt:abiversion | awk -F= '{print $$2; exit}')
+$(shell $(APK) query --fields tags --match provides $(1) | tr ' ' '\n' | grep '^openwrt:abiversion=' | cut -d= -f2)
 endef
 
 # Format packages by adding an ABI version suffix if found
 